lib/api/api.go

```diff
@@ -153,6 +156,10 @@ func (s *service) getListener(guiCfg config.GUIConfiguration) (net.Listener, err
 		if err != nil {
 			name = s.tlsDefaultCommonName
 		}
+		name, err = sanitizedHostname(name)
```

Comment:
Wouldn't punnycode encoding be easier?

Reply:
Maybe, maybe not? We'd still need a sanitization step beforehand as punycode/IDN deals with well formed (but international) domain names, and there's nothing to say that our input is well formed, and the result would look more like crap in case someone actually looked at it. 🤷